jormungandr/src/blockchain/process.rs

```diff
@@ -446,16 +446,31 @@ pub fn process_new_ref(
         .get_ref()
         .and_then(move |tip_ref| {
             if tip_ref.hash() == candidate.block_parent_hash() {
-                info!(logger, "update current branch tip");
+                info!(
+                    logger,
+                    "update current branch tip: {} -> {}",
+                    tip_ref.header().description(),
+                    candidate.header().description(),
+                );
                 A(A(tip.update_ref(candidate).map(|_| true)))
             } else {
                 match chain_selection::compare_against(blockchain.storage(), &tip_ref, &candidate) {
                     ComparisonResult::PreferCurrent => {
-                        info!(logger, "create new branch");
+                        info!(
+                            logger,
+                            "create new branch with tip {} | current-tip {}",
```

Comment:
Could be done with KV pairs instead, that would be better for JSON log parsers and the like.

Reply:
yes. this can be revised on top, but right now doesn't really matter